scala-package/core/src/main/scala/org/apache/mxnet/IO.scala

```diff
@@ -216,6 +244,30 @@ object DataBatch {
       this
     }
 
+    /**
+      * Set the dtype.
+      * @param dataDType The dtype of the data, default is Float32
+      * @param labelDType The dtype of the label, default is Int32
+      * @return this
+      */
+    def setDType(dataDType: DType, labelDType: DType): Builder = {
```

Comment:
Look at the constructor you are getting all 4 fields of DataDesc. 
```
                private val providedData: ListMap[String, Shape],
                private val providedLabel: ListMap[String, Shape],
                val dataDType: DType,
                val labelDType: DType,
                val dataLayout: String,
                val labelLayout: String)
```

Reply:
But since user always define these field as null, it will introduce crashes when we fetch the information from here, I didn't even know why we need provideData and provideLabel here.